### DISCUSSION TITLE
How does cmd argument "--preview-option" work ?

### DISCUSSION THREAD

**Discussion by patientx** [lllyasviel/Fooocus#2023] [Q&A]
I entered it like "--preview-option taesd" but app sees them as seperate things in log and doesn't seem to work at all, tried it with none or others. How do we correctly use it in arguments ?

**Comment by eddyizm** [comment]
Can you post what you seen and your log? 
Are you using the bat file or the cli directly?
What os are you using?

**Reply by patientx** [reply]
windows 10 
bat file

"python entry_with_update.py --directml --preview-option none --always-low-vram --disable-in-browser"

Tried putting it like this , --preview-option-none , then app exits meaning it doesnt accept it but if the other one was the correct one why doesnt it work ? (I just want to be able to taesd or disable it entirely without having to check the box everytime)

-------------------------------
Update succeeded.
[System ARGV] ['entry_with_update.py', '--directml', '--preview-option', 'none', '--always-low-vram', '--disable-in-browser']
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Fooocus version: 2.1.862
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Using directml with device:
Total VRAM 1024 MB, total RAM 16309 MB
Set vram state to: LOW_VRAM
Always offload VRAM
Device: privateuseone
VAE dtype: torch.float32
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --attention-split
Refiner unloaded.
model_type EPS
UNet ADM Dimension 2816

**Comment by DirkHeche** [comment]
Same problem here. Fooocus seems to be ignoring the command line option. I also found no description about what should be the result of the other options (none, fast, taesd). Is there a secret manual I have just not found yet?

**Reply by patientx** [reply]
none probably disabling it, taesd is the vae that produces images at lower quality but a lot faster /and with smaller memory if used in the final stage of actual production, fast .. I don't know :)

**Reply by eddyizm** [reply]
There may be some flags that have not been implemented yet. Someone else found one last week and I created a PR to implement it. This may be the same case. Without checking what is the intended effect?

**Reply by DirkHeche** [reply]
See below: I think the root cause may be something different.

**Comment by DirkHeche** [comment]
I checked the code base because I was trying to understand why the --preview-option could be ignored. I think the following code block in args_parser.py is preventing the command line options to be read at all:

```
if ldm_patched.modules.options.args_parsing:
    args = parser.parse_args([])
else:
    args = parser.parse_args([])
```

The command line options are only being parsed as a default. As we are giving an empty array into the parse_args function, that is being parsed and all "real" command line options are being ignored (as they are never being parsed).
